girder/tasks.py

```diff
@@ -0,0 +1,20 @@
+from girder.models.assetstore import Assetstore
+from girder.utility.progress import ProgressContext
+from girder_worker.app import app
+
+
+@app.task(queue='local')
+def importDataTask(
+    assetstore: dict,
+    parent: dict,
+    parentType: str,
+    params: dict,
+    progress: bool,
+    user: dict,
+    leafFoldersAsItems: bool
+):
+    with ProgressContext(progress, user=user, title='Importing data') as ctx:
```

Comment:
All `queue='local'` tasks can use any existing mechanisms that require database access, such as `ProgressContext`. Because these are using the `girder_worker` app, they'll also create and manage Job records automatically.